Manager/src/main/java/io/raspberrywallet/manager/modules/pin/PinModule.java

```diff
@@ -35,24 +41,34 @@ public String getHtmlUi() {
         return "<input type=\"text\" name=\"pin\">";
     }
 
-
     @Override
-    public byte[] encrypt(byte[] payload) throws RequiredInputNotFound {
-        return crypt(payload);
+    public byte[] encrypt(byte[] payload) throws RequiredInputNotFound, EncryptionException {
+        Password password = new Password(getPin());
+    
+        AESEncryptedObject<ByteWrapper> encryptedObject =
+                CryptoObject.encrypt(new ByteWrapper(payload), password);
+        
+        return SerializationUtils.serialize(encryptedObject);
```

Comment:
Could you please exaplain why do we need to involve Java Serialization  ?
https://github.com/RaspberryWallet/Backend/issues/66